src/parser.rs

```diff
@@ -537,3 +585,38 @@ fn test_environment_validation() {
     );
     assert_eq!(validate_environment("good"), Ok(()));
 }
+
+#[test]
+fn test_version_ordering() {
+    macro_rules! ver {
+        ($v:expr) => {
+            Version::parse($v).unwrap()
+        };
+    }
+
+    assert_eq!(ver!("1.0.0"), ver!("1.0.0"));
+    assert_ne!(ver!("1.1.0"), ver!("1.0.0"));
+    assert_eq!(ver!("1.0"), ver!("1.0.0"));
+    assert_eq!(ver!("1.0dev"), ver!("1.0.0-dev"));
+    assert_eq!(ver!("1.0dev"), ver!("1.0.0.0-dev"));
+    assert_ne!(ver!("1.0dev"), ver!("1.0.0.0-dev1"));
+
+    assert!(ver!("1.0dev") >= ver!("1.0.0.0-dev"));
+    assert!(ver!("1.0dev") < ver!("1.0.0.0"));
+
+    assert!(ver!("1.0.0") > ver!("1.0.0-rc1"));
+    assert!(ver!("1.0.0") >= ver!("1.0.0-rc1"));
+    assert!(ver!("1.0.0-rc1") > ver!("0.9"));
+    assert!(ver!("1.0.0+10") < ver!("1.0.0+20"));
+    assert!(ver!("1.0.0+a") < ver!("1.0.0+b"));
+
+    assert!(ver!("1.0") < ver!("2.0"));
+    assert!(ver!("1.1.0") < ver!("10.0"));
+    assert!(ver!("1.1.0") < ver!("1.1.1"));
+    assert!(ver!("1.1.0.1") < ver!("1.1.0.2"));
+    assert!(ver!("1.1.0.1") > ver!("1.1.0.0"));
+    assert!(ver!("1.1.0.1") > ver!("1.0.0.0"));
+    assert!(ver!("1.1.0.1") > ver!("1.0.42.0"));
+
+    assert!(ver!("1.0+abcd") < ver!("1.0+abcde"));
+}
```

Comment:
Nit: I would add these just to confirm behavior

```suggestion
assert!(ver!("1.0.0-alpha.beta") < ver!("1.0.0-beta"));
assert!(ver!("1.0.0-beta") < ver!("1.0.0-beta.2"));
assert!(ver!("1.0.0-alpha.2") < ver!("1.0.0-alpha.11"));
assert!(ver!("1.0.0-beta.11") < ver!("1.0.0-rc.1"));
}
```